pyproject.toml

```diff
@@ -87,3 +81,93 @@ task-tags = ["Example", "todo", "TODO", "FIXME"]
 
 [tool.codespell]
 ignore-words-list = 'edn,juxt'
+
+[build-system]
+requires = ["setuptools>=65", "setuptools_scm[toml]>=7.1"]
+
+[tool.setuptools]
+packages = ["rocky", "octopoes", "keiko", "boefjes", "mula", "bytes"]
+
+[tool.setuptools_scm]
+write_to = "_version.py"
+
+[project]
+name = "openkat"
+version = "0.0.1.dev1"
+description = "OpenKAT coordination package"
+authors = [
+    { name = "MinVWS", email = "maintainer@openkat.nl" }
+]
+license = { text = "EUPL" }
+readme = "README.rst"
+requires-python = ">=3.8"
+
+[project.optional-dependencies]
+dev = [
+    "pre-commit==3.3.2",
+    "sphinx==6.1.3",
+    "sphinx_rtd_theme==1.2.0",
+    "sphinxcontrib-mermaid==0.8.1",
+    "myst-parser==1.0.0"
+]
+
+common = []
+common-dev = [
+    "pytest==7.2.2",
+    "exceptiongroup==1.1.1", # needed for pytest on Python <3.11
+    "httpx==0.23.3",
+]
```

Comment:
I think we should only put direct dependencies in pyproject. An easy way to figure out direct dependencies is the pipreqs tool, but that might miss some things such as pytest fixtures.

The following aren't a direct dependency and can be removed from pyproject as far as I can see:
- certifi
- dnspython
- httpx
- typing-extension

Reply:
Do we still version pin dependencies of our dependencies some way if we only pin our direct dependencies?